### PR TITLE
test user credential updates and delete

### PR DESCRIPTION
**Pull request by cendhu**

### CODE REVIEW

File: test/setup/cluster_setup.go [hunk 1]
```diff
@@ -386,45 +387,59 @@ func (c *Cluster) createUsersCryptoMaterials() error {
...
+	pemUserKeyFile, err := os.Create(userKeyPath)
+	if err != nil {
+		return err
+	}
+	if _, err = pemUserKeyFile.Write(pemUserKey); err != nil {
+		return err
+	}
+	if err = pemUserKeyFile.Close(); err != nil {
+		return err
+	}
```
Comment (tock-ibm) [inline test/setup/cluster_setup.go:430]: you can use `os.WriteFile` t do all this in shorter way.

File: test/setup/cluster_setup.go [hunk 2]
```diff
@@ -386,45 +387,59 @@ func (c *Cluster) createUsersCryptoMaterials() error {
...
-		}
+	return nil
+}
 
-		userKeyPath := path.Join(c.testDirAbsPath, "users", user+".key")
-		pemUserKeyFile, err := os.Create(userKeyPath)
+func (c *Cluster) CreateOrUpdateUserCerts(user string, keyPair tls.Certificate) error {
+	var err error
+	if reflect.ValueOf(keyPair).IsZero() {
+		keyPair, err = tls.X509KeyPair(c.rootCAPemCert, c.caPrivKey)
```
Comment (tock-ibm) [inline test/setup/cluster_setup.go:399]: This pattern is confusing... in other packages it seems it is always "Update", whereas internally in the test package it is always create. Can we provide two methods? they may map to an internal "ugly" method, but passing in an empty key as flag for Update is a bit obscure for my taste.

File: test/setup/cluster_setup.go (hunk 1 above)
Comment (cendhu) [reply test/setup/cluster_setup.go]: Done

File: test/setup/cluster_setup.go (hunk 2 above)
Comment (cendhu) [reply test/setup/cluster_setup.go]: Done

File: test/setup/cluster_setup.go [hunk 3]
```diff
@@ -380,51 +388,68 @@ func (c *Cluster) createUsersCryptoMaterials() error {
...
+	// if err != nil {
+	// 	return err
+	// }
+	// _, err = pemUserCertFile.Write(pemUserCert)
+	// if err != nil {
+	// 	return err
+	// }
+	// if err = pemUserCertFile.Close(); err != nil {
+	// 	return err
+	// }
```
Comment (tock-ibm) [inline test/setup/cluster_setup.go:425]: please remove

File: test/setup/cluster_setup.go [hunk 4]
```diff
@@ -380,51 +388,68 @@ func (c *Cluster) createUsersCryptoMaterials() error {
...
+	// pemUserKeyFile, err := os.Create(userKeyPath)
+	// if err != nil {
+	// 	return err
+	// }
+	// if _, err = pemUserKeyFile.Write(pemUserKey); err != nil {
+	// 	return err
+	// }
+	// if err = pemUserKeyFile.Close(); err != nil {
+	// 	return err
+	// }
```
Comment (tock-ibm) [inline test/setup/cluster_setup.go:440]: please remove

**Comment by tock-ibm** [comment]
all looks good except for a couple of commented out code that was left behind

File: test/setup/cluster_setup.go (hunk 3 above)
Comment (cendhu) [reply test/setup/cluster_setup.go]: Done

File: test/setup/cluster_setup.go (hunk 4 above)
Comment (cendhu) [reply test/setup/cluster_setup.go]: Done

**Review by tock-ibm — APPROVED** [review]

**Pull request merged**